KLSadd_insertion/src/updateChapters.py

```diff
@@ -164,185 +400,237 @@ def referencePlacer(chap, references, p, kls, refII, refIII, specref):
+        # store the file as a string
+        addendum = add.readlines()
+        kls_list_all = new_keywords(addendum, kls_list_full)[0]
+        # Makes sections look like other sections
+        sorter_check = new_keywords(addendum, kls_list_full)[1]
+
+        for item in addendum:
+            addendum[addendum.index(item)] = item.replace("\\eqref{","\\eqref{KLS Addendum: ")
+
+        for word in addendum:
+            if "paragraph{" in word:
+                lenword = len(word) - 1
+                temp = word[0:word.find("{") + 1] + "\\bf KLS Addendum: " + word[word.find("{") + 1: lenword]
+                addendum[addendum.index(word)] = temp
+            if "subsubsection*{" in word:
+                lenword = len(word) - 1
+                addendum[addendum.index(word)] = word[0:word.find("{") + 1] + "\\bf KLS Addendum: " + \
+                word[word.find("{") + 1: lenword]
+        index = 0
+        indexes = []
+
+        # Designates sections that need stuff added
+        # get the index
+        startindex = 9999
+        for word in addendum:
+            index += 1
+            if "." in word and "\\subsection*{" in word:
+                if "9." in word:
+                    chap_nums.append(9)
+                    name = word[word.find("{") + 1: word.find("}")]
+                    math_people.append(name + "#")
+                if "14." in word:
+                    chap_nums.append(14)
+                    name = word[word.find("{") + 1: word.find("}")]
+                    math_people.append(name + "#")
+                if name == "9.1 Wilson":
+                    startindex = addendum.index(word)
+                    print startindex
+                indexes.append(index-1)
+            if "paragraph{" in word and index > startindex-1:
+                klsaddparas.append(index-1)
+            if "subsub" in word and index > startindex-1:
+                klsaddparas.append(index - 1)
+            if "\subsection*{" in word and index > startindex-1:
+                klsaddparas.append(index - 1)
+            if "\\renewcommand{\\refname}{Standard references}" in word:
+                klsaddparas.append(index-1)
+                indexes.append(index - 1)
+        print(indexes)
+        # now indexes holds all of the places there is a section
+        # using these indexes, get all of the words in between and add that to the paras[]
+        paras = []
+        for i in range(len(indexes)-1):
+            box = ''.join(addendum[indexes[i]: indexes[i+1]-1])
+            paras.append(box)
+
+
+        # paras now holds the paragraphs that need to go into the chapter files, but they need to go in the appropriate
+        # section(like Wilson, Racah, Hahn, etc.) so we use the mathPeople variable
+        # we can use the section names to place the relevant paragraphs in the right place
+
+        # as of 2/8/16 the paragraphs will go before the References paragraph of the relevant section
```

Comment:
Date references in comments are not necessary when there is a version control system that shows when things were added